synapseclient/client.py

```diff
@@ -302,28 +309,47 @@ def login(self, email=None, password=None, apiKey=None, sessionToken=None, remem
         """
         Authenticates the user using the given credentials (in order of preference):
 
-        - supplied synapse user name (or email) and password
-        - supplied email and API key (base 64 encoded)
-        - supplied session token
-        - supplied email and cached API key
-        - most recent cached email and API key
-        - email in the configuration file and cached API key
-        - email and API key in the configuration file
-        - email and password in the configuraton file
-        - session token in the configuration file
-
-        :param email:      Synapse user name (or an email address associated with a Synapse account)
+        #. supplied arguments
+
+        #. .synapseConfig file
+
+        #. cached credentials from rememberMe=True
+
+        Valid combinations of authentication information:
```

Comment:
please explain more about the scope of where these values must be set. Can they be set one value at one place and another value at another place?